File: python/python/ch7/check_performance.py
import pydash
import toolz as tz
from .PerformanceChecker import CheckerTimeType, log_checker, map_between_checker, start_checker, stop_checker

check_performance = lambda name, func: tz.compose(
    log_checker(CheckerTimeType.ms),
    stop_checker,
    map_between_checker(func),
    start_checker(False, 0)
)(name)

File: javascript/ch7/checkPerformance.js
const fp = require('lodash/fp');
const { logChecker, mapBetweenChecker, startChecker, stopChecker } = require('./PerformanceChecker');

const checkPerformance = (name, func) => fp.compose(
  logChecker('ms'),
  stopChecker,
  mapBetweenChecker(func),
  startChecker(false, 0),
)(name);

module.exports = checkPerformance;